src/Operations.js

```diff
@@ -822,18 +825,22 @@ export function zipWithFactory(keyIter, zipper, iters) {
       let steps;
       if (!isDone) {
         steps = iterators.map(i => i.next());
-        isDone = steps.some(s => s.done);
+        if (zipAll) isDone = steps.every(s => s.done);
+        else isDone = steps.some(s => s.done);
       }
+
```

Comment:
This is the critical change in logic, as @leebyron [pointed out](https://github.com/facebook/immutable-js/pull/460/files#r29642099) in an earlier pull request.